pkgs/tools/security/creddump/default.nix

```
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitLab, python2, python2Packages }:
+
+python2Packages.buildPythonApplication rec {
+  pname = "creddump";
+  version = "0.3";
+
+  src = fetchFromGitLab {
+    owner = "kalilinux";
+    repo = "packages/creddump";
+    # url-encoding workaround: https://github.com/NixOS/nixpkgs/issues/65796#issuecomment-517829019
+    rev = "debian%2F${version}-1kali2"; # %2F = urlquote("/")
+    sha256 = "0r3rs2hggsvv619l3fh3c0jli6d3ryyj30ni3hz0nz670z5smzcf";
+  };
+
+  # No setup.py is available
+  dontBuild = true;
+  doCheck = false;
+  pythonPath = with python2Packages; [ pycrypto ];
+  buildInput = [ python2 ];
+
+  installPhase = ''
+    mkdir -p $out/bin
```

Change this here and in the other commands.

```suggestion
    mkdir -p ${placeholder "out"}/bin
```

---

Ok adjusted :+1:  
Out of curiosity: why is this better?

---

TBH I'm not sure how to explain this I just had that comment in the past. I think the idea is that you get the actual content of what's in `$out` inside these shell hooks so it's 1 less thing the shell evaluates and 1 more thing Nix does..

---

I just found a [thread](https://discourse.nixos.org/t/what-is-the-difference-between-placeholder-out-and-out/4862) which claims to avoid `placeholder` if possible, as it is designed to be used for edge cases in which you need to reference `$out` outside of classic places like `buildPhase` or `installPhase`. However, I cannot really find a good reason for that claim inside the [referenced documentation](https://nixos.org/nix/manual/#ssec-relnotes-2.0) :(

Anyways I also do not really see a downside using `placeholder`, so I do not mind applying it here :+1: